Fix asm tokens leaking horrendously because they used Vector and PVector

Rot127 opened the pull request:
<!-- Filling this template is mandatory -->

**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository.
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style).
- [ ] I've documented every `RZ_API` function and struct this PR changes.
- [ ] I've added tests that prove my changes are effective (required for changes to `RZ_API`).
- [ ] I've updated the [Rizin book](https://github.com/rizinorg/book) with the relevant information (if needed).

**Detailed description**

<!-- Explain the **details** for making this change. Is a new feature implemented? What existing problem does the pull request solve? How does the pull request solve these issues? Please provide enough information so that others can review your pull request. -->

I must have missed some places back then, when converting to pvector. And it leaked like there is no tomorrow.
This should fix it now.

https://github.com/rizinorg/rizin/pull/5015 is still valid, but I probably ran the wrong build (without LSAN). Hence thought it fixed it.

**Test plan**

<!-- What steps should the reviewer take to test your pull request? Demonstrate the code is solid. Example: The exact commands you ran and their output, screenshots/videos. This is your time to re-check that everything works and that you covered all the edge cases -->

All green

**Closing issues**

<!-- put "closes #XXXX" in your comment to auto-close the issue that your PR fixes (if any). -->

...

Rot127 committed:
Fix asm tokens leaking horrendously because they used Vector and PVector.

Rot127 commented:
Still missed some somewhere.

Rot127 committed:
Fix some resource leaks related to asm tokens

Rot127 committed:
Fix some resource leaks of the hexagon module error cases.

Rot127 committed:
Fix possible NULL ptr dereference

Rot127 committed:
Fix leak of asm tokens by fun fini on RzAsmOp.

The RzAsmOp passed here could be a previously initialized one.
With allocated asm_tokens. Doing memset just leaks those tokens.

Rot127 committed:
Simplify the copy_asm_op function

Rot127 committed:
Always fini RzAsmOp before initializeing it.

notxvilka reviewed (changes requested):
1. Please rebase
2. There is a double free somewhere?
```
[XX] 828 ms   db/esil/ppc_32 long ESIL ppc-32
RZ_NOPLUGINS=1 /usr/bin/rizin -escr.utf8=0 -escr.color=0 -escr.interactive=0 -eflirt.sigdb.load.system=false -esearch.show_progress=false -eflirt.sigdb.load.home=false -N -qc 'e asm.arch=ppc
e asm.bits=32
e cfg.bigendian=true
(pi bytes;wx ${bytes};pi 1;pie 1)
.(pi 4200fff0)
' =
-- stdout
--- expected
+++ actual
@@ --1,4 +-1,3 @@
 bc 0x10, lt, 0xfffffff0
-0x00000000 1,ctr,-=,$z,!,?{,0xfffffff0,pc,=,},

-- stderr
free(): invalid pointer

-- exit status: -6

```

Rot127 committed:
Set asm_toks to NULL after freeing

Rot127 committed:
Fix leak in asm_bf

Rot127 committed:
Fix leaks in test_token

Rot127 committed:
Initialize all RzAsmOps with 0 and free the asm_toks on init.

In practice rz_asm_op_init() is the best place to free the asm_toks reliably.
Simply because this is always called in the old code, before an instruction is disassembled.
Only in fini() is not enough, because there are too many place it is not called, leaking the tokens.

notxvilka commented on librz/arch/aop.c:
```diff
@@ -16,6 +16,7 @@ RZ_API void rz_asm_op_init(RZ_NULLABLE RzAsmOp *op) {
 	if (!op) {
 		return;
 	}
+	rz_asm_token_string_free(op->asm_toks);
```
Interesting solution.

Rot127 replied:
Yes, this is bad. But the to cite the commit messaage:

> In practice rz_asm_op_init() is the best place to free the asm_toks reliably.
Simply because this is always called in the old code, before an instruction is disassembled.
> Only in fini() is not enough, because there are too many place it is not called, leaking the tokens.

wargio replied:
This looks incorrect. We expect this structure to possibly contain garbage.

notxvilka reviewed (approved):
LGTM. The only doubt I have is that `_free()` function in the `_init()`

wargio reviewed (changes requested):
This looks wrong

Rot127 committed:
Fresh mind, new attempt. Fini asmop at all occurrences of rz_asm_disassemble

notxvilka reviewed (approved):
This approach is certainly better!

notxvilka commented:
@wargio take a look again please
